manifests/repos.pp

```diff
@@ -60,19 +63,47 @@
 
 ) {
   if $create_repos {
+    $parts = split($kubernetes_version, '[.]')
+    $minor_version = "${parts[0]}.${parts[1]}"
+    $_repos = $kubernetes_apt_repos ? {
+      undef   => '',
+      default => $kubernetes_apt_repos
+    }
     case $facts['os']['family'] {
       'Debian': {
-        $codename = fact('os.distro.codename')
         apt::source { 'kubernetes':
-          location => pick($kubernetes_apt_location,'https://apt.kubernetes.io'),
-          repos    => pick($kubernetes_apt_repos,'main'),
-          release  => pick($kubernetes_apt_release,'kubernetes-xenial'),
-          key      => {
-            'id'     => pick($kubernetes_key_id,'A362B822F6DEDC652817EA46B53DC80D13EDEF05'),
-            'source' => pick($kubernetes_key_source,'https://packages.cloud.google.com/apt/doc/apt-key.gpg'),
-          },
+          location => pick($kubernetes_apt_location,"https://pkgs.k8s.io/core:/stable:/v${minor_version}/deb"),
+          release  => pick($kubernetes_apt_release, '/'),
+          repos    => $_repos,
         }
 
+        if $kubernetes_apt_location =~ String[1] {
+          Apt::Source<| title == 'kubernetes' |> {
+            key => {
+              'id'     => $kubernetes_key_id,
+              'source' => $kubernetes_key_source,
+            }
+          }
+        } else {
+          # For pkgs.k8s.io use GPG siging key
+          $_keyring = '/usr/share/keyrings/kubernetes-apt-keyring.gpg'
+          # TODO: Switch to apt::keyring once supported by puppetlabs-apt
+          # see: https://github.com/puppetlabs/puppetlabs-apt/pull/1128
+          archive { '/tmp/kubernetes-apt-keyring.gpg':
+            source          => "https://pkgs.k8s.io/core:/stable:/v${minor_version}/deb/Release.key",
+            extract         => true,
+            extract_path    => '/usr/share/keyrings',
+            extract_command => 'gpg --dearmor < %s > kubernetes-apt-keyring.gpg',
+            creates         => $_keyring,
+          }
+
+          Apt::Source<| title == 'kubernetes' |> {
+            keyring  => $_keyring,
+            require  => Archive['/tmp/kubernetes-apt-keyring.gpg'],
+          }
```

Comment:
As I wrote earlier, it would be very nice to switch to modern `apt::keyring`. But we should make incremental changes. Adding dependency on `puppetlabs-apt >= 9.2.0` means effectively bumping dependency on `puppetlabs-stdlib` from `>= 4.25.0` to `>= 9.0.0`. This is rather huge step for the mankind, especially done in a single innocent PR called "Swich repos...". 

Depending on puppet code base setup this might mean upgrading many (10 or 40) modules just because of this minor change. I would prefer to release this "old but working version" as a `8.1.0` version. And then change module dependencies in separate PR, while bumping the module version e.g. to `9.0.0`.

Reply:
As mentioned in #681, the actual `puppetlabs/stdlib` requirement is currently `>= 8.4.0`. So the change wouldn't be that huge.